### PR TITLE
[IE] setBatchSize through reshape

### PR DESCRIPTION
**Pull request by jane-intel**

### CODE REVIEW

File: inference-engine/tests_deprecated/functional/shared_tests/network_tests/network_i8.hpp [hunk 1]
```diff
@@ -341,8 +341,10 @@ class smoke_NetworkClassifyTest : public TestsCommon, public TestsCommonFunc, pu
             network = ie.ReadNetwork(p.model(), p.weights());
         }
 
-        if (batch_size != 1)
+        if (batch_size != 1) {
+            auto implNet = dynamic_cast<InferenceEngine::details::CNNNetworkImpl *>(&((InferenceEngine::ICNNNetwork&)network));
             network.setBatchSize(batch_size);
+        }
```
Comment (ilya-lavrenov) [inline inference-engine/tests_deprecated/functional/shared_tests/network_tests/network_i8.hpp:347]: from my point of view, we need code like this:
```
        if (batch_size != 1) {
            if (network.getFunction()) {
                auto cnnNetworkImpl = std::make_shared<details::CNNNetworkImpl>(network);
                network = CNNNetwork(cnnNetworkImpl);
            } 
            network.setBatchSize(batch_size);
        }
```

File: inference-engine/src/inference_engine/cnn_network_ngraph_impl.cpp [hunk 2]
```diff
@@ -433,46 +434,38 @@ StatusCode CNNNetworkNGraphImpl::serialize(const std::string& xmlPath, const std
...
         for (const auto &parameter : original_parameters) {
-            if (parameter->get_partial_shape().is_dynamic())
-                THROW_IE_EXCEPTION << "Cannot setBatch! Network contains inputs with dynamic shapes!";
-            std::vector<size_t> shape = parameter->get_shape();
-            origShapes[parameter->get_friendly_name()] = shape;
+            auto pshape = parameter->get_partial_shape();
+            if (pshape.rank().is_static() && (pshape.rank().get_length() == 0 || pshape.rank().get_length() == 1 || pshape.rank().get_length() == 3))
+                continue;  // WA: for speech recognition and scalar layouts (copy-past from CNNNetwork)
+            if (pshape.is_dynamic()) {
+
```
Comment (ilyachur) [inline inference-engine/src/inference_engine/cnn_network_ngraph_impl.cpp:456]: nit: redundant line

**Review by ilyachur — COMMENTED** [review]
Can we add tests to check behaviour of setBatch/getBatch methods for nGraph function with dynamic shapes?

**Comment by jane-intel** [comment]
> Can we add tests to check behaviour of setBatch/getBatch methods for nGraph function with dynamic shapes?

Done

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Comment by ilyachur** [comment]
@jane-intel Please fix CI.

**Comment by jane-intel** [comment]
> @jane-intel Please fix CI.

There is a related MR to product-config -- pre-commit scope should be corrected? so I'm waiting for merge

**Comment by ilya-lavrenov** [comment]
Can we restore `setBatchSize` as is, but all changes move to `reshapeBatch` / `setBatchSizeReshape`? After, I think it can be merged

**Comment by jane-intel** [comment]
I suggest not to touch this methods in this release.

**Comment by jane-intel** [comment]
Closing in favour of https://github.com/openvinotoolkit/openvino/pull/2597 which is already merged

**Pull request closed without merging**